kernelci/config/base.py

```diff
@@ -24,15 +24,64 @@
 # Common classes for all config types
 #
 
-class YAMLObject:
+class YAMLConfigObject(yaml.YAMLObject):
+    """Base class with helper methods to handle configuration YAML data
+
+    This class contains methods to help constructing configuration objects from
+    YAML data.  Then each subclass should implement its standard `to_yaml()`
+    method to be able to dump the whole configuration hierarchy back to YAML.
+    """
+
+    @classmethod
+    def load_from_yaml(cls, config, **kwargs):
+        """Load the YAML configuration
+
+        Load the YAML configuration passed as a *config* data structure with a
+        given *name*.  This method should return an instance of a _YAMLObject
+        subclass.
+        """
+        yaml_attributes = cls._get_yaml_attributes()
+        kwargs.update(cls._kw_from_yaml(config, yaml_attributes))
+        return cls(**kwargs)
+
+    @classmethod
+    def _kw_from_yaml(cls, data, attributes):
+        """Create some keyword arguments based on a YAML dictionary
+
+        Return a dictionary suitable to be used as Python keyword arguments in
+        an object constructor using values from some YAML *data*.  The
+        *attributes* are a list of keys to look up from the *data* and convert
+        to a dictionary.  Keys that are not in the YAML data are simply omitted
+        from the returned keywords, relying on default values in object
+        constructors.
+
+        """
+        return {
+            k: v for k, v in ((k, data.get(k))for k in attributes)
+            if v is not None
+        } if data else dict()
+
+    @classmethod
+    def _get_yaml_attributes(cls):
+        """Get a set of YAML attribute names
+
+        Get a set object with all the YAML configuration attribute names for
+        the configuration class.  This can be used to make keyword arguments
+        when creating a configuration object as well as when serialising it
+        back to YAML.
+        """
+        return set()
+
+
+class _YAMLObject:
```

Comment:
I don't undertstand the reason behind renaming `YAMLObject` to `_YAMLObject`. 

Usually in Python any name starting with an underscore is meant for internal use. It's clearly specified for method, attribute and function names. I think it's not for class names. In the standard library there is a `_thread` module which is basically provides a low-level API for `threading` module which is supposed to be used in public. 
In this case `_YAMLObject` is used outside of this module which may be a bit confusing.

Reply:
This is to make it clear `_YAMLObject` is not the standard `yaml.YAMLObject`.

Reply:
We can call it `NotRealYAMLObject` or anything else if you want, but it has to be a different name as the real `yaml.YAMLObject` standard class is now starting to be used.

Reply:
Ok, I get it, readability matters. I can see couple of ways dealing with it:

- Rename to something like `KciYAMLObject` (maybe a bit too many acronyms here)
- Leave it to namespaces - I think the base approach to make it clearly visible that it's not the `yaml.YAMLObject`

```python
from kernelci.config import base

class MyClass(base.YAMLObject)
```
It can be also left as is. All in all there seem to be no convention defined for `_Class` in PEP8.

Reply:
The goal is actually to remove `_YAMLObject` by converting all the classes to `yaml.YAMLObject`.  There's a PR for the build configs already, and follow-up ones will cover the rest of them.